### PR TITLE
PIM-4895: Fix redirect after mass edit unpublished products

### PR DESCRIPTION
**Pull request by pierallard**
| Q                  | A
| ------------------ | ---
| Migration script?  | n
| Behats             | y
| Specs & Static     | n
| Changelog updated  | n

### CODE REVIEW

**Comment by nidup** [comment]
please do PR on 1.4, the 1.5 is now the master and this issue is a bug

File: src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php [hunk 1]
```diff
@@ -283,4 +279,22 @@ protected function getQueryParams()
...
+     */
+    protected function getRouteFromGridName($gridName)
+    {
+        $route = 'pim_enrich_product_index';
+
+        if ('family-grid' === $gridName) {
+            $route = 'pim_enrich_family_index';
+        }
+
+        return $route;
```
Comment (ClementGautier) [inline src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php]: I think using early return would be more readable here:

```
if ('family-grid' === $gridName) {
    return 'pim_enrich_family_index';
}

return 'pim_enrich_product_index';
```

File: src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php [hunk 2]
```diff
@@ -283,4 +279,22 @@ protected function getQueryParams()
 
         return $params;
     }
+
+    /**
+     * Return the route to follow after a performed action
+     *
+     * @param $gridName
```
Comment (ClementGautier) [inline src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php]: missing type

**Comment by ClementGautier** [comment]
I don't see the difference here, how do it fix the bug ?

File: src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php (hunk 2 above)
Comment (ClementGautier) [inline src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php]: use `@inherited` as the function belongs to the CE

File: src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php (hunk 1 above)
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php]: I think that we should use a switch here. This is the real purpose of a switch. Your code tend to say that product_index is more important/different from family_index

File: src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php [hunk 3]
```diff
@@ -226,11 +226,7 @@ public function performAction($operationAlias)
                 sprintf('pim_enrich.mass_edit_action.%s.launched_flash', $operationAlias)
             );
 
-            $route = 'pim_enrich_product_index';
-
-            if ('family-grid' === $gridName) {
-                $route = 'pim_enrich_family_index';
-            }
+            $route = getRouteFromGridName($gridName);
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php]: $this->getRouteblabla

File: src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php (hunk 1 above)
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Controller/MassEditActionController.php]: There was a switch before and 2 or 3 peoples ask me to change for this solution. I'm pretty ok with @ClementGautier. If there is more than 2 solutions use switch, but with  solutions early return seems better.

**Pull request closed without merging**